testsuite/MDAnalysisTests/analysis/test_density.py

```diff
@@ -147,3 +151,29 @@ def test_density_from_Universe_update_selection(self):
             self.selections['dynamic'],
             self.references['dynamic']['meandensity'],
             update_selections=True)
+
+class TestGridImport(TestCase):
+
+    def setUp(self):
+        sys.modules.pop('MDAnalysis.analysis.density', None)
+
+    @block_import('gridData')
+    def test_absence_griddata(self):
+        # if gridData package is missing an ImportError should be raised
+        # at the module level of MDAnalysis.analysis.density
+        with assert_raises(ImportError):
+            import MDAnalysis.analysis.density
+
+    def test_presence_griddata(self):
+        # no ImportError exception is raised when gridData is properly
+        # imported by MDAnalysis.analysis.density
+
+        # mock gridData in case there are testing scenarios where
+        # it is not available
```

Comment:
No my reasoning is that either we demand dependencies in 'analysis` are optional or we make them all a requirement. I don't see a reason for cherry picking here. I never had problems to install scipy or matplotlib different linux or mac machines (using pip/homebrew/...). Also we exclude some of the most popular scientific packages here that really try to make the installation easy.